lua/entities/cfc_stinger_missile.lua

```diff
@@ -0,0 +1,397 @@
+                dmgSound = "cfc_stinger_impactflesh"
+
+            end
+        elseif hitEnt:IsPlayer() then
+            -- this ends up getting added with the blastdamage, doesn't need to be too strong
+            dmgAmount = 75 * stingerDirectHitPlayerMul:GetFloat()
+            dmgSound = "cfc_stinger_impactflesh"
+        end
+
+        return dmgAmount, dmgSound
+    end
+
+    function ENT:HitEntity( hitEnt )
+        if IsValid( hitEnt ) then
+            local Pos = self:GetPos()
+            -- hit simfphys car instead of simfphys wheel
+            if hitEnt.GetBaseEnt and IsValid( hitEnt:GetBaseEnt() ) then
+                hitEnt = hitEnt:GetBaseEnt()
+            end
+
+            local effectdata = EffectData()
+                effectdata:SetOrigin( Pos )
+                effectdata:SetNormal( -self:GetForward() )
+            util.Effect( "manhacksparks", effectdata, true, true )
+
+            local dmgAmount, dmgSound = self:GetDirectHitDamage( hitEnt )
+            dmgAmount = dmgAmount * stingerDmgMulCvar:GetFloat()
+
+            local dmginfo = DamageInfo()
+                dmginfo:SetDamage( dmgAmount )
+                dmginfo:SetAttacker( IsValid( self:GetAttacker() ) and self:GetAttacker() or self )
+                dmginfo:SetDamageType( DMG_DIRECT )
+                dmginfo:SetInflictor( self )
+                dmginfo:SetDamagePosition( Pos )
+                dmginfo:SetDamageForce( self:GetForward() * dmgAmount * 500 )
+            hitEnt:TakeDamageInfo( dmginfo )
+
+            sound.Play( dmgSound, Pos, 140 )
+
+        end
+
+        self:Detonate()
+    end
+
+    function ENT:BreakMissile()
+        if not self:GetDisabled() then
+            self:SetDisabled( true )
+
+            local pObj = self:GetPhysicsObject()
+
+            if IsValid( pObj ) then
+                pObj:EnableGravity( true )
+                self:PhysWake()
+                self:EmitSound( "Missile.ShotDown" )
+            end
+        end
+    end
+
+    function ENT:Detonate()
+        local dmgMul = stingerDmgMulCvar:GetFloat()
+
+        local Inflictor = self:GetInflictor()
+        local Attacker = self:GetAttacker()
+
+        local ExplodePos = self:WorldSpaceCenter()
+
+        local effectdata = EffectData()
+            effectdata:SetOrigin( self:GetPos() )
+        util.Effect( "Explosion", effectdata )
+
+        self:EmitSound( "vehicles/airboat/pontoon_impact_hard1.wav", 100, 50, 0.5 )
+        self:EmitSound( "Explo.ww2bomb" )
+
+        self:Remove()
+
+        timer.Simple( 0, function()
+            local FallbackDamager = Entity( 0 )
+            Inflictor = IsValid( Inflictor ) and Inflictor or FallbackDamager
+            Attacker = IsValid( Attacker ) and Attacker or FallbackDamager
+
+            util.BlastDamage( Inflictor, Attacker, ExplodePos, 200 * dmgMul, 150 * dmgMul )
+        end )
+    end
+
+    function ENT:OnTakeDamage( dmginfo )
+        if dmginfo:GetDamageType() ~= DMG_AIRBOAT then return end
+
```

Comment:
Damage types are bitflags, you shouldn't compare directly.
```suggestion
        if not dmginfo:IsDamageType( DMG_AIRBOAT ) then return end
```